src/main/java/org/thoughtcrime/securesms/ConversationFragment.java

```diff
@@ -339,6 +340,12 @@ protected void setCorrectMenuVisibility(Menu menu) {
             boolean showReplyPrivately = chat.isMultiUser() && !messageRecord.isOutgoing() && canReply;
             menu.findItem(R.id.menu_context_reply_privately).setVisible(showReplyPrivately);
             menu.findItem(R.id.menu_add_to_home_screen).setVisible(messageRecord.getType() == DcMsg.DC_MSG_WEBXDC);
+
+            boolean saved = messageRecord.getSavedMsgId() != 0;
+            MenuItem toggleSave = menu.findItem(R.id.menu_toggle_save);
+            toggleSave.setVisible(messageRecord.canSave() && !chat.isSelfTalk());
+            toggleSave.setIcon(saved? R.drawable.baseline_bookmark_border_24 : R.drawable.baseline_bookmark_24);
```

Comment:
let's do that the other way round, it feels wrong showing action but not state here.

also as the other  icons are less bold.

cmp. to the proxy settings, where we also have "filled" for enabled.

```suggestion
            toggleSave.setIcon(saved? R.drawable.baseline_bookmark_24 : R.drawable.baseline_bookmark_border_24);
```